internal/impl/io/input_http_server.go

```diff
@@ -348,11 +348,13 @@ input:
 
 // HTTTPInputMiddlewareMeta is a custom middleware type for HTTP server inputs that adds metadata to messages.
 type HTTTPInputMiddlewareMeta func(*http.Request) (map[string]any, error)
+type HTTTPInputMiddlewareMetaConstructor func(conf *service.ParsedConfig) (HTTTPInputMiddlewareMeta, error)
 
 // RegisterCustomHTTPServerInput registers a custom HTTP server input with a given name and optional middleware.
-func RegisterCustomHTTPServerInput(name string, middleware HTTTPInputMiddlewareMeta) {
+func RegisterCustomHTTPServerInput(name string, mdlWareConst HTTTPInputMiddlewareMetaConstructor, extraSpec *service.ConfigField) {
+	spec := hsiSpec().Field(extraSpec)
```

Comment:
Let's add a nil check looks like [field will panic](https://github.com/redpanda-data/benthos/blob/7da18c5fb096a3607f9f70c06614742b24ae938e/public/service/config.go#L412-L434)

```suggestion
    if extraSpec != nil {
	   spec := hsiSpec().Field(extraSpec)
	}
```